### PR TITLE
Add e2e test for unsummarized ops nack recovery

### PR DESCRIPTION
**Pull request by kian-thompson**
We currently don't have any tests verifying our recovery flows when we hit 10k un-summarized op issues. The only way we have is to analyze production data. Thus, we need to add an e2e test to help verify that mitigations we introduce actually solve the issues.

[AB#6917](https://dev.azure.com/fluidframework/internal/_workitems/edit/6917)

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts [hunk 1]
```diff
@@ -371,8 +371,8 @@ describeCompat("SharedInterval", "NoCompat", (getTestObjectProvider, apis) => {
 			intervals.add({ start: 0, end: 2 });
 			assertIntervals([{ start: 0, end: 2 }]);
 
-			for (let j = 0; j < 10; j++) {
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts:374]: Why is this needed?

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts [hunk 2]
```diff
@@ -402,6 +411,99 @@ describeCompat("Summaries", "NoCompat", (getTestObjectProvider) => {
 		await provider.ensureSynchronized();
 		await summaryCollection.waitSummaryAck(container.deltaManager.lastSequenceNumber);
 	});
+
+	async function getNackProm(container: IContainer) {
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:415]: nit: I haven't seen `Prom` anywhere else in the repo, probably best to stick with `Promise` or `P`

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts [hunk 3]
```diff
@@ -402,6 +411,99 @@ describeCompat("Summaries", "NoCompat", (getTestObjectProvider) => {
...
+		return new Promise<void>((resolve, reject) => {
+			const callback = (_reason, error) => {
+				try {
+					assert.strictEqual(error?.statusCode, 429);
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:419]: Opinion: The flow try-assert-catch-reject is kind of confusing, I started to put a comment down calling out that the asserts won't affect the Promise but then saw what's going on.

But I'm guessing your point is to get the `assert` library to generate the `Error` for you.  I was going to suggest just doing `if (...) { reject(...); }` but it's up to you.

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts [hunk 4]
```diff
@@ -402,6 +411,99 @@ describeCompat("Summaries", "NoCompat", (getTestObjectProvider) => {
...
+			container.deltaManager.on("disconnect", callback);
+		});
+	}
+
+	it("Can summarize after hitting nack on unsummarized ops", async function () {
+		if (provider.driver.type !== "local") {
+			this.skip();
+		}
+		const stringId = "sharedStringKey";
+		const registry: ChannelFactoryRegistry = [[stringId, SharedString.getFactory()]];
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:440]: Is this typically necessary?  In e2e tests I've worked on I haven't specified the registry

**Review by markfields — APPROVED** [review]

File: packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts (hunk 1 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts:374]: I reduced the number of ops this test sends since it was the only one sending more than 200 unsummarized ops (a setting I changed in local server to enable the test I added)

**Comment by alexvy86** [comment]
@kian-thompson can we make this configurable with an environment variable or some other way? I think it broke the performance memory tests, where [now many complain about container closes due to too many reconnects with no progress syncing local ops](https://dev.azure.com/fluidframework/internal/_build/results?buildId=240482&view=logs&j=d042b387-7692-5f02-a062-ee417cc37fe5&t=c9404925-d8e2-5bd7-253e-b1cab12afc6d&l=96).

**Comment by kian-thompson** [comment]
> @kian-thompson can we make this configurable with an environment variable or some other way? I think it broke the performance memory tests, where [now many complain about container closes due to too many reconnects with no progress syncing local ops](https://dev.azure.com/fluidframework/internal/_build/results?buildId=240482&view=logs&j=d042b387-7692-5f02-a062-ee417cc37fe5&t=c9404925-d8e2-5bd7-253e-b1cab12afc6d&l=96).

@alexvy86 The tricky part is that the driver is determined when spinning up the e2e tests, so there's currently not an easy way to have this single test be different. Does increasing the unsummarized ops to 10k help with the performance tests?
https://github.com/microsoft/FluidFramework/blob/main/packages/test/test-drivers/src/localServerTestDriver.ts#L33

**Comment by alexvy86** [comment]
I don't know for sure which part of this change is the culprit, if any. If I get some time I'll try to test if increasing that back to 10k gets them working again.
